### PR TITLE
[Doc] advanced example for simulating proppant slot test

### PR DESCRIPTION
**Pull request by jhuang2601**
Proppant model has been validated by comparing numerical results with experimental data. 
This PR documents the validation example by using refactored proppant solver (#1559).

### CODE REVIEW

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 1]
```diff
@@ -0,0 +1,292 @@
+.. _ExampleProppantTest:
+
+
+####################################################
+Simulate Proppant Slot Test
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:5]:
```suggestion
Proppant Slot Test
```

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+####################################################
+
+
+**Context**
+
+In this example, a simulation is built up to model proppant slot test. In this way, the implemented proppant model is validated by comparing numerical results with the corresponding experimental data. Furthermore, this calibrated proppant model can allow field engineers to customize stimulation design and optimize field operations in multiple engineering aspects `(Huang et al., 2021) <https://onepetro.org/ARMAUSRMS/proceedings/ARMA21/All-ARMA21/ARMA-2021-1248/467980>`__.
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:11]:
```suggestion
In this example, a simulation is built up to model a proppant slot test. In this way, the implemented proppant model is validated by comparing numerical results with the corresponding experimental data. Furthermore, this calibrated proppant model can allow field engineers to customize stimulation design and optimize field operations in multiple engineering aspects `(Huang et al., 2021) <https://onepetro.org/ARMAUSRMS/proceedings/ARMA21/All-ARMA21/ARMA-2021-1248/467980>`__.
```

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+
+**Input file**
+
+This example uses no external input files and everything required is
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:16]:
```suggestion
This example uses no external input files and everything is
```

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
...hun et al. (2020) <https://www.sciencedirect.com/science/article/pii/S0920410519309441>`__ conducted slot tests on proppant transport with slickwater. As shown below, a 4 ft X 1 ft slot with 0.3 in (0.00762 m) gap width was constructed. Three fluid inlets with 0.5 in inner diameter were placed at the right side of the slot, which were three inches away from each other. One outlet was place on the top side to allow pressure relief. The other one was located on the left side acting as a fluid sink. In their test, to resemble a slickwater fracturing treatment, the proppant concentration was kept at 1.5 ppg and the viscosity of carrying fluid was approximately 1 cp. The slurry was mixed well and then injected into the flow channel at a constant injection rate (6 gpm). A simulation case with the same settings is built up to mimic this slot test. A vertical and impermeable fracture surface is assumed in this case, which eliminates the effect of fracture plane inclination and fluid leakoff. A static fracture with uniform aperture (0.3 in) is defined and fracture propagation is not involved. 30/50 mesh proppant is injected via the three inlets and is flowed through the slot for 30 seconds.
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:28]: Maybe in this description we could use the GEOSX unit system everywhere (meters, Pa.s, etc)?

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+To simulate proppant transport phenomenon, a proppant solver based on the assumption of multi-component single phase flow is used in this example. Proppant concentration and distribution within the slot are numerically calculated by solving the equations of proppant transport in hydraulic fractures. These numerical predictions are then compared with the corresponding testing results `(Chun et al., 2020)  <https://www.sciencedirect.com/science/article/pii/S0920410519309441>`__ to check the accuracy of modelling results.
+
+
+In this example, we focus our attention on the ``Solver`` tags, the ``Constitutive`` tags, and the ``FieldSpecifications`` tags.
+
+------------------------------
+Mesh
+------------------------------
+
+Following figure shows the generated mesh that is used for solving this problem.
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:49]:
```suggestion
The following figure shows the mesh used for solving this problem.
```

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+.. _problemSketchFig:
+.. figure:: mesh.png
+   :align: center
+   :width: 500
+   :figclass: align-center
+
+   Generated mesh
+
+
+We use the internal mesh generator ``InternalWellbore`` to create a computational domain.
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:61]:
```suggestion
We use the internal mesh generator ``InternalMesh`` to create a computational domain. 
```

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+
+
+.. literalinclude:: ../../../../../../src/coreComponents/physicsSolvers/multiphysics/benchmarks/Proppant-transport-validation-case.xml
+  :language: xml
+  :start-after: <!-- SPHINX_PROPPANT_MESH -->
+  :end-before: <!-- SPHINX_PROPPANT_MESH_END -->
+
+
+---------------------------
+Proppant Transport solver
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:74]:
```suggestion
Proppant transport solver
```
to be consistent with the capitalization used in the other advanced examples

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+
+In this example, let us focus on the coupling solver.
+This solver (``FlowProppantTransport``) describe the coupling process between proppant and flow transport within the ``Fracture`` region. In this way, the two single-physics solvers (``ProppantTransport`` and ``SinglePhaseFVM``) will be simultaneously called and executed for solving this proppant transport problem.
+
+
+------------------------------
+Constitutive laws
+------------------------------
+
+For this slot test, 30/50 mesh proppant was injected via the three inlets and flowing through the slot for 30 seconds.
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:120]:
```suggestion
For this slot test, 30/50 mesh proppant is injected via the three inlets and flowing through the slot for 30 seconds. 
```
and I suggest explaining what 30/50 mesh proppant is (I admit I do not know what it means)

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 9]
```diff
@@ -0,0 +1,292 @@
...
+In this example, let us focus on the coupling solver.
+This solver (``FlowProppantTransport``) describe the coupling process between proppant and flow transport within the ``Fracture`` region. In this way, the two single-physics solvers (``ProppantTransport`` and ``SinglePhaseFVM``) will be simultaneously called and executed for solving this proppant transport problem.
+
+
+------------------------------
+Constitutive laws
+------------------------------
+
+For this slot test, 30/50 mesh proppant was injected via the three inlets and flowing through the slot for 30 seconds. 
+The viscosity of carrying fluid was 1 cp to resemble slickwater fracturing. In this example, the solid and fluid materials are named as ``sand`` and ``water`` respectively.
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:121]:
```suggestion
The viscosity of carrying fluid is 0.001 Pa.s to resemble slickwater fracturing. In this example, the solid and fluid materials are named as ``sand`` and ``water`` respectively. 
```

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 10]
```diff
@@ -0,0 +1,292 @@
...
+Initial and boundary conditions
+-----------------------------------------------------------
+
+The next step is to specify fields, including:
+
+  - The initial value (fracture aperture, fluid pressure and proppant concentration within the fracture have to be initialized)
+  - The boundary conditions (fluid pressure and proppant concentration at fluid inlets and outlets)
+ 
+These boundary conditions are set up through the ``FieldSpecifications`` section.
+At a constant injection rate, the slurry was equally flowing into the open channel through three inlets.
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:143]:
```suggestion
At a constant injection rate, the slurry is equally flowing into the open channel through three inlets.
```

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 11]
```diff
@@ -0,0 +1,292 @@
...
+Note: For static (non-propagating) fracture problems, the fields ``ruptureState`` and ``elementAperture`` should be provided in the initial conditions.
+
+
+
+The parameters used in the simulation are summarized in the following table.
+
++------------------+----------------------------------+------------------+---------------+
+| Symbol           | Parameter                        | Unit             | Value         |
++==================+==================================+==================+===============+
+| :math:`d_p`      | Proppant Diameter                | [mm]             | 0.45          |
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:160]:
```suggestion
| :math:`d_p`      | Proppant Diameter                | [m]             | 0.00045          |
```

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 12]
```diff
@@ -0,0 +1,292 @@
...
++------------------+----------------------------------+------------------+---------------+
+| :math:`c_s`      | Max Fraction of Proppant         | [-]              | 0.62          |
++------------------+----------------------------------+------------------+---------------+
+| :math:`\rho_f`   | Fluid Density                    | [kg/m^3]         | 1000          |
++------------------+----------------------------------+------------------+---------------+
+| :math:`\mu_f`    | Fluid Viscosity                  | [Pa*s]           | 0.001         |
++------------------+----------------------------------+------------------+---------------+
+| :math:`\rho_p`   | Proppant Density                 | [kg/m^3]         | 2550          |
++------------------+----------------------------------+------------------+---------------+
+| :math:`\lamda_s` | Hindered Settling Coefficient    | [-]              | 4.5           |
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:174]:
```suggestion
| :math:`\lambda_s` | Hindered Settling Coefficient    | [-]              | 4.5           |
```

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 13]
```diff
@@ -0,0 +1,292 @@
...
+
+
+------------------------------------------------------------------
+To go further
+------------------------------------------------------------------
+
+
+**Feedback on this example**
+
+This concludes the example on simulating proppant slot test.
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:285]:
```suggestion
This concludes the example on simulating a proppant slot test.
```

File: src/coreComponents/physicsSolvers/multiphysics/benchmarks/Proppant-transport-validation-case.xml [hunk 14]
```diff
@@ -0,0 +1,328 @@
...
+      solidNames="{ fractureFilling }"
+      permeabilityNames="{ fracturePerm }">
+      <NonlinearSolverParameters
+        newtonTol="1.0e-6"
+        newtonMaxIter="8"
+        lineSearchAction="None"
+        newtonMinIter="0"/>
+      <LinearSolverParameters
+        solverType="gmres"
+	preconditionerType="amg"
```
Comment (francoishamon) [inline src/coreComponents/physicsSolvers/multiphysics/benchmarks/Proppant-transport-validation-case.xml:57]: Indentation

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 15]
```diff
@@ -0,0 +1,292 @@
...
+- the coupling solver (``FlowProppantTransport``) that will bind the two single-physics solvers above, which is named as ``FlowProppantTransport``
+
+.. literalinclude:: ../../../../../../src/coreComponents/physicsSolvers/multiphysics/benchmarks/Proppant-transport-validation-case.xml
+  :language: xml
+  :start-after: <!-- SPHINX_PROPPANT_COUPLEDSOLVER -->
+  :end-before: <!-- SPHINX_PROPPANT_COUPLEDSOLVER_END -->
+
+
+In this example, let us focus on the coupling solver.
+This solver (``FlowProppantTransport``) describe the coupling process between proppant and flow transport within the ``Fracture`` region. In this way, the two single-physics solvers (``ProppantTransport`` and ``SinglePhaseFVM``) will be simultaneously called and executed for solving this proppant transport problem.
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:113]:
```suggestion
This solver (``FlowProppantTransport``) describes the coupling process between proppant and flow transport within the ``Fracture`` region. In this way, the two single-physics solvers (``ProppantTransport`` and ``SinglePhaseFVM``) are sequentially called to solve the sub-problems (proppant transport and pressure problem, respectively) involved in this test case.
```

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 16]
```diff
@@ -0,0 +1,292 @@
...
+
+- the single-phase flow solver, a solver of type ``SinglePhaseProppantFVM`` called here ``SinglePhaseFVM``,
+
+.. literalinclude:: ../../../../../../src/coreComponents/physicsSolvers/multiphysics/benchmarks/Proppant-transport-validation-case.xml
+  :language: xml
+  :start-after: <!-- SPHINX_PROPPANT_FLOWSOLVER -->
+  :end-before: <!-- SPHINX_PROPPANT_FLOWSOLVER_END -->
+
+
+- the coupling solver (``FlowProppantTransport``) that will bind the two single-physics solvers above, which is named as ``FlowProppantTransport``
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:104]:
```suggestion
- the coupling solver (``FlowProppantTransport``) that binds the two single-physics solvers above, which is named as ``FlowProppantTransport``
```

File: src/coreComponents/physicsSolvers/multiphysics/benchmarks/Proppant-transport-validation-case.xml [hunk 17]
```diff
@@ -0,0 +1,328 @@
+<?xml version="1.0" ?>
```
Comment (francoishamon) [inline src/coreComponents/physicsSolvers/multiphysics/benchmarks/Proppant-transport-validation-case.xml:1]: Should be remove the example here: https://github.com/GEOSX/GEOSX/blob/develop/examples/proppantTransport/Proppant-transport-validation-case.xml
and just keep the xml you are adding in this PR?

**Review by francoishamon — APPROVED** [review]
Looks great, thanks @jhuang2601

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 18]
```diff
@@ -0,0 +1,293 @@
...
+| :math:`a`        | Fracture Aperture                | [m]              | 0.00762       |
++------------------+----------------------------------+------------------+---------------+
+| :math:`Q`        | Injection Rate                   | [m^3/s]          | 0.0003785     |
++------------------+----------------------------------+------------------+---------------+
+
+---------------------------------
+Inspecting results
+---------------------------------
+
+The following figure shows the modelling prediction of proppant distribution at t=10s and t=30s, which are compared with the experiments in `(Chun et al., 2020)  <https://www.sciencedirect.com/science/article/pii/S0920410519309441>`__. Due to proppant settling in low viscosity fluid, a heterogeneous proppant distribution is obtained, which evolves with injection time. Three different zones (immobile proppant bed, suspended proppant and clean fluid) are visually identified for both the presented experiment and simulation.
```
Comment (sytuannguyen) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:192]:
```suggestion
The following figure shows the modelling prediction of proppant distribution at 10 s and 30 s, which are compared with the experiments in `(Chun et al., 2020)  <https://www.sciencedirect.com/science/article/pii/S0920410519309441>`__. Due to proppant settling in low viscosity fluid, a heterogeneous proppant distribution is obtained, which evolves with injection time. Three different zones (immobile proppant bed, suspended proppant and clean fluid) are visually identified for both the presented experiment and simulation. 
```

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 19]
```diff
@@ -0,0 +1,293 @@
...
+       totalCellNum = 96 * 24
+       font = {'size'   : 12}
+       matplotlib.rc('font', **font)
+ 
+       fig, ax = plt.subplots(2,2,figsize=(12, 8))
+
+       ax[0,0].plot(eTime, eLength, 'ro', label='Experiment-30/50')
+       ax[0,0].plot(dTime, (dLength-offSize)/totalLength, lw=5, alpha=0.5, color= 'lime', label='GEOSX-30/50 mesh', linestyle='-')
+       ax[0,0].set_ylabel('Normalized Bank Length', multialignment='center', weight="bold")
+       ax[0,0].set_xlabel('Time (second)', weight="bold")
```
Comment (sytuannguyen) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:237]:
```suggestion
       ax[0,0].set_xlabel('Time (s)', weight="bold")
```

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 20]
```diff
@@ -0,0 +1,293 @@
...
+       ax[0,0].set_xlim(0, 40)
+       ax[0,0].set_ylim(0.0, 1.2)
+       ax[0,0].legend(frameon=False,loc='lower right',fontsize=12)
+       ax[0,0].text(.01,.92,'(a)', horizontalalignment='left', transform=ax[0,0].transAxes, size = 15, weight = 'bold')
+
+
+       ax[0,1].plot(eTime, eArea, 'ro', label='Experiment-30/50')
+       ax[0,1].plot(dTime, dArea/totalCellNum, lw=5, alpha=0.5, color= 'lime', label='GEOSX-30/50 mesh', linestyle='-')
+       ax[0,1].set_ylabel('Normalized Bank Area', multialignment='center', weight="bold")
+       ax[0,1].set_xlabel('Time (second)', weight="bold")
```
Comment (sytuannguyen) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:247]:
```suggestion
       ax[0,1].set_xlabel('Time (s)', weight="bold")
```

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 21]
```diff
@@ -0,0 +1,293 @@
...
+       ax[0,1].set_xlim(0, 40)
+       ax[0,1].set_ylim(0.0, 0.5)
+       ax[0,1].legend(frameon=False,loc='lower right',fontsize=12)
+       ax[0,1].text(.01,.92,'(b)', horizontalalignment='left', transform=ax[0,1].transAxes, size = 15, weight = 'bold')
+
+
+       ax[1,0].plot(esTime, esArea, 'ro', label='Experiment-30/50')
+       ax[1,0].plot(sTime, sArea/totalCellNum, lw=5, alpha=0.5, color= 'lime', label='GEOSX-30/50 mesh', linestyle='-')
+       ax[1,0].set_ylabel('Normalized Suspended Proppant Area', multialignment='center', weight="bold")
+       ax[1,0].set_xlabel('Time (second)', weight="bold")
```
Comment (sytuannguyen) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:257]:
```suggestion
       ax[1,0].set_xlabel('Time (s)', weight="bold")
```

File: src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst [hunk 22]
```diff
@@ -0,0 +1,293 @@
...
+       ax[1,0].set_xlim(0, 40)
+       ax[1,0].set_ylim(0.0, 0.6)
+       ax[1,0].legend(frameon=False,loc='upper right',fontsize=12)
+       ax[1,0].text(.01,.92,'(c)', horizontalalignment='left', transform=ax[1,0].transAxes, size = 15, weight = 'bold')
+
+
+       ax[1,1].plot(eTime, eArea+esArea, 'ro', label='Experiment-30/50')
+       ax[1,1].plot(dTime, (dArea+sArea)/totalCellNum, lw=5, alpha=0.5, color= 'lime', label='GEOSX-30/50 mesh', linestyle='-')
+       ax[1,1].set_ylabel('Normalized Total Propped Area', multialignment='center', weight="bold")
+       ax[1,1].set_xlabel('Time (second)', weight="bold")
```
Comment (sytuannguyen) [inline src/docs/sphinx/advancedExamples/validationStudies/proppantSlotTest/Example.rst:267]:
```suggestion
       ax[1,1].set_xlabel('Time (s)', weight="bold")
```

**Review by sytuannguyen — APPROVED** [review]
Very well/clearly explained, well done!

**Review by joshua-white — APPROVED** [review]

**Pull request merged**